### ISSUE TITLE
Issue with Expo 44

### ISSUE THREAD

**Issue by cmosboss** [wodin/expo-zip-example#2]
Not sure if you ever ran into this issue before. Trying to add this to a new Expo 44 project. Everything seems to be working except when I try and actually run the unzip function on my custom development client I get this error.

```Error: [TypeError: null is not an object (evaluating 'RNZipArchive.unzip')]```

When configuring this did you do anything other than expo install react-native-zip-archive, publish your new development client, and then try and use the code?

**Comment by wodin** [comment]
Hi @cmosboss 

Expo SDK 45 was released today. I'll upgrade this app and give it a go and let you know how it goes.

**Comment by wodin** [comment]
I messed up the upgrade to begin with, but fixed it and it works fine. My Android dev client is here if you want to give it a try:

https://expo.dev/accounts/wodin/projects/zip-test/builds/9a91e09c-2513-489c-84c0-14a89a944745

The error you're getting makes it sound like you don't have the native code in your dev client. It's the same sort of error I would expect from Expo Go, since that doesn't include the native code from `react-native-zip-archive`.

Is it possible you built the dev client before installing `react-native-zip-archive`?

Basically what I did was as you describe (except I didn't "publish" anything. Just installed the dev client on my phone):
* expo init
* expo install expo-dev-client
* yarn add react-native-zip-archive
* eas build ...
* Install the resulting dev client on a device/emulator/simulator

Then you should be able to edit the JS to use `react-native-zip-archive`.

**Comment by wodin** [comment]
Here's a "preview" Simulator build that works fine for me too:

https://expo.dev/accounts/wodin/projects/zip-test/builds/69443994-5d2e-4bfd-88a2-147276cfb15f

**Comment by tacitomv** [comment]
I just got the same block and found this issue here. I'm using Expo Go... I had in mind it would work :D

I'm new to this expo thing, am I expecting something I shouldn't?

**Comment by wodin** [comment]
Hi @tacitomv 

`react-native-zip-archive` needs some native code that is not included in Expo Go, so it will not work in Expo Go.
But you can build a custom dev client (a.k.a. a development build.) to use instead of Expo Go.
A custom dev client is basically a custom version of Expo Go that is built using your dependencies instead of just what's in the Expo SDK. So if you need `react-native-zip-archive`, then you can build that into your dev client.

See: https://docs.expo.dev/development/introduction/
